cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -307,6 +375,100 @@ subroutine evp (dt)
       enddo  ! iblk
       !$TCXOMP END PARALLEL DO
 
+      if (grid_system == 'CD') then
+
+      !$TCXOMP PARALLEL DO PRIVATE(iblk,ilo,ihi,jlo,jhi,this_block)
+      do iblk = 1, nblocks
+
+      !-----------------------------------------------------------------
+      ! more preparation for dynamics on N grid
+      !-----------------------------------------------------------------
+
+         this_block = get_block(blocks_ice(iblk),iblk)         
+         ilo = this_block%ilo
+         ihi = this_block%ihi
+         jlo = this_block%jlo
+         jhi = this_block%jhi
+
+         call dyn_prep2 (nx_block,             ny_block,             & 
+                         ilo, ihi,             jlo, jhi,             &
+                         icellt(iblk),         icelln(iblk),         & 
+                         indxti      (:,iblk), indxtj      (:,iblk), & 
+                         indxni      (:,iblk), indxnj      (:,iblk), & 
+                         aiN       (:,:,iblk), nmass     (:,:,iblk), & 
+                         nmassdti  (:,:,iblk), fcorN_blk  (:,:,iblk), & 
+                         nmask     (:,:,iblk),                       & 
+                         uocnN     (:,:,iblk), vocnN     (:,:,iblk), & 
+                         strairxN  (:,:,iblk), strairyN  (:,:,iblk), & 
+                         ss_tltxN  (:,:,iblk), ss_tltyN  (:,:,iblk), &  
+                         icetmask  (:,:,iblk), icenmask  (:,:,iblk), & 
+                         fmN       (:,:,iblk), dt,                   & 
+                         strtltxN  (:,:,iblk), strtltyN  (:,:,iblk), & 
+                         strocnxN  (:,:,iblk), strocnyN  (:,:,iblk), & 
+                         strintxN  (:,:,iblk), strintyN  (:,:,iblk), & 
+                         taubxN    (:,:,iblk), taubyN    (:,:,iblk), & 
+                         waterxN   (:,:,iblk), wateryN   (:,:,iblk), & 
+                         forcexN   (:,:,iblk), forceyN   (:,:,iblk), & 
+                         stressp_1 (:,:,iblk), stressp_2 (:,:,iblk), & 
+                         stressp_3 (:,:,iblk), stressp_4 (:,:,iblk), & 
+                         stressm_1 (:,:,iblk), stressm_2 (:,:,iblk), & 
+                         stressm_3 (:,:,iblk), stressm_4 (:,:,iblk), & 
+                         stress12_1(:,:,iblk), stress12_2(:,:,iblk), & 
+                         stress12_3(:,:,iblk), stress12_4(:,:,iblk), & 
+                         uvelN_init (:,:,iblk), vvelN_init (:,:,iblk), &
+                         uvelN      (:,:,iblk), vvelN      (:,:,iblk), &
+                         TbN       (:,:,iblk))
+
+      enddo  ! iblk
+      !$TCXOMP END PARALLEL DO
+
+      !$TCXOMP PARALLEL DO PRIVATE(iblk,ilo,ihi,jlo,jhi,this_block)
+      do iblk = 1, nblocks
```

Comment:
@dabail10 these two loops could be fusioned, right ? any reason why you added separate loops ?

Reply:
This was fixed in PR #27